float32 x_ref
float32 theta_ref
